app/src/components/avatar.rs

```diff
@@ -0,0 +1,133 @@
+use crate::{
+    components::avatar_event_bus::{AvatarEventBus, Response},
+    infra::common_component::{CommonComponent, CommonComponentParts},
+};
+use anyhow::{bail, Result};
+use graphql_client::GraphQLQuery;
+use serde::{Deserialize, Serialize};
+use yew::prelude::*;
+use yew_agent::{Bridge, Bridged};
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "queries/get_user_avatar.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+pub struct GetUserAvatar;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct AvatarData(String);
+
+impl AvatarData {
+    pub fn new(data: String) -> Self {
+        AvatarData(data)
+    }
+}
+
+pub struct ShowAvatar {
+    common: CommonComponentParts<Self>,
+    avatar: Option<AvatarData>,
+    _producer: Box<dyn Bridge<AvatarEventBus>>,
+}
+
+/// State machine describing the possible transitions of the component state.
+/// It starts out by fetching the user's details from the backend when loading.
+pub enum Msg {
+    /// Received the user details response, either the user data or an error.
+    UserAvatarResponse(Result<get_user_avatar::ResponseData>),
+    Update(Response),
+}
+
+#[derive(yew::Properties, Clone, PartialEq, Eq)]
+pub struct Props {
+    pub username: String,
+    pub width: i32,
+    pub height: i32,
+}
+
+impl CommonComponent<ShowAvatar> for ShowAvatar {
+    fn handle_msg(
+        &mut self,
+        ctx: &Context<Self>,
+        msg: <Self as Component>::Message,
+    ) -> Result<bool> {
```

Comment:
How about `mut avatar_changed = false;` and when you update the avatar to a different value than is already stored, you set it to true?

That way you don't have multiple happy path returns, you can keep a single return at the end.